packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -209,22 +217,18 @@ class AndroidSdk {
 }
 
 class AndroidSdkVersion implements Comparable<AndroidSdkVersion> {
-  AndroidSdkVersion(this.sdk, {
-    this.platformVersionName,
+  AndroidSdkVersion._(this.sdk, {
+    this.sdkLevel,
+    this.platformName,
     this.buildToolsVersion,
   });
 
   final AndroidSdk sdk;
-  final String platformVersionName;
+  final int sdkLevel;
+  final String platformName;
   final Version buildToolsVersion;
-  String get buildToolsVersionName => buildToolsVersion.toString();
 
-  int get sdkLevel {
-    if (_namedVersionMap.containsKey(platformVersionName))
-      return _namedVersionMap[platformVersionName];
-    else
-      return int.parse(platformVersionName.substring('android-'.length));
-  }
+  String get buildToolsVersionName => buildToolsVersion.toString();
```

Comment:
Given that we depend on this being non-null, consider adding asserts in the ctor. Looks like we defend against that on ll. 199-200, but probably worth the extra paranoia.